utilities/blob_db/blob_compaction_filter.cc

```diff
@@ -102,13 +286,14 @@ BlobIndexCompactionFilterFactory::CreateCompactionFilter(
   }
   assert(current_time >= 0);
 
-  BlobCompactionContext context;
-  blob_db_impl_->GetCompactionContext(&context);
-
-  return std::unique_ptr<CompactionFilter>(new BlobIndexCompactionFilter(
-      context, static_cast<uint64_t>(current_time), statistics_));
+  return FilterTraits<Filter>::Create(
+      blob_db_impl_, static_cast<uint64_t>(current_time), statistics_);
```

Comment:
Is the reason creating a template to reuse these 10 lines above? I feel like it ends up with complicating the code, rather than simplifying it. I'm personally not good at understanding code with templates. I believe many teammate are the same. Also, Google C++ Style recommends to be cautious about it too: https://google.github.io/styleguide/cppguide.html#Template_metaprogramming

Reply:
This is not actually TMP, just plain old generic programming. I personally don't like any kind of code duplication, and templates are often a great way of eliminating that but like you said, it's only ~10 lines of code so I don't think it's a big deal either way.

BTW, we should really have a conversion about the Google Style Guide at some point :)